test/Fail/Issue5657.err

```diff
@@ -1,3 +1,5 @@
-Issue5657.agda:11.11-12: error: [ShouldBePi]
-Set should be a function type, but it isn't
+Issue5657.agda:11.11-12: error: [CannotApply]
+Expression used as function but does not have function type:
+  expr: .A
```

Comment:
Here we are pretending that a postfix projection `.A` is an expression by itself. Ideally the error for the test case (in https://github.com/agda/agda/blob/master/test/Fail/Issue5657.agda) should mention that the projection is already applied to its principal argument, but in the meantime we should avoid making the error message more confusing than it is.

Reply:
Yes, indeed, we should formulate the function differently when the function part is a postfix projection.

To my defense, it can hardly get worse than the original error "Set should be a function type", which is utter nonsense. :-D
So I feel I have improved the situation, while there is room for further improvement.